src/main/java/graphql/nadel/SchemaSourceFactory.java

```diff
@@ -0,0 +1,18 @@
+package graphql.nadel;
+
+
+import com.atlassian.braid.Link;
+import com.atlassian.braid.SchemaNamespace;
+import com.atlassian.braid.SchemaSource;
+import graphql.nadel.dsl.ServiceDefinition;
+import graphql.schema.idl.TypeDefinitionRegistry;
+
+import java.util.List;
+
+/**
+ * Factory for schema source.
+ */
+public interface SchemaSourceFactory {
```

Comment:
same as above: I don't think we should expose Braid types, rather some Nadel specifc "SchemaSource"

Reply:
I do not agree, if you see Nadel constructor it already takes GraphQLRemoteRetrieverFactory that provides GraphQLRemoteRetriever which is braid type (and it was here already).  It is not enough however for stitching service and we need more general way of creating schema source that's why I've added SchemaSourceFactory interface. 
I do not like adding some abstractions over braid types as they will be mapped one to one internally to braid types, this is additional cost at no benefits at all (we do not plan to move away from braid in foreseeable future), also I believe that we will have to expose internals of braid to allow support for transformations that are not likely to be implemented here as they will be to specific.